src/stream/src/executor/backfill/upstream_table/external.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::catalog::{Schema, TableId};
+use risingwave_common::util::row_serde::*;
+use risingwave_common::util::sort_util::OrderType;
+use risingwave_connector::source::external::{ExternalTableReaderImpl, SchemaTableName};
+
+// pub type HummockStorageType = impl StateStore + AsHummockTrait;
+/// This struct represents an external table to be read during backfill
+pub struct ExternalStorageTable {
+    /// Id for this table.
+    table_id: TableId,
+
+    /// The normalized name of the table, e.g. `dbname.schema_name.table_name`.
+    table_name: String,
+
+    schema_name: String,
+
+    table_reader: ExternalTableReaderImpl,
+
+    /// The schema of the output columns, i.e., this table VIEWED BY some executor like
+    /// RowSeqScanExecutor.
+    /// todo: the schema of the external table defined in the CREATE TABLE DDL
+    schema: Schema,
+
+    /// Used for serializing and deserializing the primary key.
+    pk_serializer: OrderedRowSerde,
```

Comment:
It doesn't need to do any serializing and deserializing, right?

I think here we only need `Vec<OrderType>`